src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/InTableFilter.java

```diff
@@ -0,0 +1,106 @@
+package io.stargate.sgv2.jsonapi.service.operation.filters.table;
+
+import static com.datastax.oss.driver.api.querybuilder.QueryBuilder.bindMarker;
+
+import com.datastax.oss.driver.api.core.metadata.schema.ColumnMetadata;
+import com.datastax.oss.driver.api.querybuilder.relation.OngoingWhereClause;
+import com.datastax.oss.driver.api.querybuilder.relation.Relation;
+import com.datastax.oss.driver.api.querybuilder.term.Term;
+import io.stargate.sgv2.jsonapi.exception.ErrorCodeV1;
+import io.stargate.sgv2.jsonapi.exception.catchable.MissingJSONCodecException;
+import io.stargate.sgv2.jsonapi.exception.catchable.ToCQLCodecException;
+import io.stargate.sgv2.jsonapi.exception.catchable.UnknownColumnException;
+import io.stargate.sgv2.jsonapi.service.cqldriver.executor.TableSchemaObject;
+import io.stargate.sgv2.jsonapi.service.operation.builder.BuiltCondition;
+import io.stargate.sgv2.jsonapi.service.operation.filters.table.codecs.JSONCodecRegistries;
+import io.stargate.sgv2.jsonapi.service.operation.query.TableFilter;
+import io.stargate.sgv2.jsonapi.service.operation.query.TableFilterAnalyzedUsage;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+/** API filter $in and $nin against table column. */
+public class InTableFilter extends TableFilter {
+
+  private final List<Object> arrayValue;
+
+  private final Operator operator;
+
+  public enum Operator {
+    IN,
+    // TODO NIN
+    NIN
+  }
+
+  public InTableFilter(Operator operator, String path, List<Object> arrayValue) {
+    super(path);
+    this.arrayValue = arrayValue;
+    this.operator = operator;
+  }
+
+  @Override
+  public <StmtT extends OngoingWhereClause<StmtT>> StmtT apply(
+      TableSchemaObject tableSchemaObject,
+      StmtT ongoingWhereClause,
+      List<Object> positionalValues) {
+    List<Term> bindMarkers = new ArrayList<>();
+
+    // TODO this codec part we won't do in the operation level? refer to insertOne
+    //    TableInsertValuesContainer
+    for (Object inValue : arrayValue) {
+      try {
+        var codec =
+            JSONCodecRegistries.DEFAULT_REGISTRY.codecToCQL(
+                tableSchemaObject.tableMetadata(), getPathAsCqlIdentifier(), inValue);
+        positionalValues.add(codec.toCQL(inValue));
+        bindMarkers.add(bindMarker());
+      } catch (UnknownColumnException e) {
```

Comment:
use ApiException errors , we should copy the ones we have int he DOCUMENT scope for unknown column and unsupported type into the FILTER scope

Reply:
convert UNKNOWN_TABLE_COLUMNS to ERROR V2.
There are other v1 usages, different types, will refactor them in following ticket.

Reply:
both of those codes are in the error.yaml file here https://github.com/stargate/data-api/blob/main/src/main/resources/errors.yaml#L123

Reply:
created ticket so this does not get missed https://github.com/stargate/data-api/issues/1488

feels sloppy to change 1 and not the other below.

Reply:
removed MULTIPLE_ID_FILTER, FIELDS_LIMIT_VIOLATION.

Added following new v2 codes, to resolve [#1488]
```
  # FILTER request errors: filter against unknown columns
  - scope: FILTER
    code: UNKNOWN_TABLE_COLUMNS
    title: Unknown table columns in filter
    body: |-
      Only columns defined in the table schema can be filtered.
      
      The table ${keyspace}.${table} defines the columns: ${allColumns}.
      The request included the following unknown columns: ${unknownColumns}.
      
      Resend the request using only defined columns.
  # FILTER request errors: filter against a column type that is not supported
  - scope: FILTER
    code: UNSUPPORTED_COLUMN_TYPES
    title: Unsupported table column data types in filter
    body: |-
      Only supported column types can be filtered
      
      The table ${keyspace}.${table} defines the columns: ${allColumns}.
      The request included the following columns that have unsupported data types: ${unsupportedColumns}.
      
      Resend the request using only supported column types.  
```